packages/components/addon/components/hds/form/character-count/index.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+const ID_PREFIX = 'character-count-';
+const NOOP = () => {};
+
+export default class HdsFormCharacterCountIndexComponent extends Component {
+  // The current number of characters in the associated input
+  @tracked currentLength;
+  inputControl = document.getElementById(this.args.controlId);
+
+  /**
+   * @param maxLength
+   * @type {number}
+   * @default null
+   * @description The maximum number of characters allowed.
+   */
+  get maxLength() {
+    let { maxLength } = this.args;
+    return parseInt(maxLength);
+  }
+
+  /**
+   * @param minLength
+   * @type {number}
+   * @default null
+   * @description The minimum number of characters allowed.
+   */
+  get minLength() {
+    let { minLength } = this.args;
+    return parseInt(minLength);
+  }
+
+  /**
+   * @param remainingLength
+   * @type {number}
+   * @default null
+   * @description The remaining number of characters.
+   */
+  get remainingLength() {
```

Comment:
see comment above about having two distinct arguments/values for the two possible differences (`min - curr` and `max - curr`)